runtime/onert/core/src/ir/train/TrainableGraph.cc

```diff
@@ -125,9 +126,71 @@ const ITrainableOperation &TrainableGraph::operation(OperationIndex index) const
   return dynamic_cast<const ITrainableOperation &>(_graph.operations().at(index));
 }
 
+void TrainableGraph::assertValidTopologicalOrder(std::vector<ir::OperationIndex> order,
+                                                 bool is_forward) const
+{
+  if (!is_forward)
+    std::reverse(order.begin(), order.end());
+
+  const std::string order_type = is_forward ? "forward" : "backward";
+
+  auto compare = [](const ir::OperationIndex &i, const ir::OperationIndex &j) -> bool {
+    return i.value() < j.value();
+  };
+
+  std::map<ir::OperationIndex, uint32_t, decltype(compare)> position(compare);
+  for (uint32_t p = 0; p < order.size(); ++p)
+  {
+    auto index = order[p];
+    if (position.count(index))
```

Comment:
(Optional)
For readability, how about using `find` instead of using `count` like below?
```suggestion
    if (position.find(index) != position.end())
```

Additionally, it would be nice to add a TODO comment like "Replace it with `contains` since C++20"

Reply:
Updated.